kasa/tapo/tapodevice.py

```diff
@@ -0,0 +1,157 @@
+"""Module for a TAPO device."""
+import base64
+import logging
+from datetime import datetime, timedelta, timezone
+from typing import Any, Dict, Optional, Set
+
+from ..aesprotocol import TPLinkAes
+from ..exceptions import AuthenticationException
+from ..smartdevice import SmartDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AttributeDict(dict):
+    """Class to allow a dict to be accessed as if keys are variables."""
+
+    __getattr__ = dict.__getitem__
+    __setattr__ = dict.__setitem__  # type: ignore[assignment]
+    __delattr__ = dict.__delitem__  # type: ignore[assignment]
```

Comment:
I'm not sure if we want to have this type of "hack" around, I'd rather prefer defensive programming with the help of `foo_dict.get("some_key")` to avoid crashing on unavailable information.

Reply:
Yes I did this just to quickly be able to pickup your `TapoPlug` that was using the TAPO library.  I've now removed this.